File: Updog.Client/src/vote/domain/vote.ts
import { VoteResourceType } from './vote-resource-type';
import { VoteDirection } from './vote-direction';

/**
 * An upvote, or downvote on a post or comment.
 */
export class Vote {
    /**
     * Create a new vote.
     * @param resouceType The type of resource being voted on.
     * @param resourceId The ID of the resource.
     * @param direction The direction of the vote.
     */
    public constructor(
        public resouceType: VoteResourceType,
        public resourceId: number,
        public direction: VoteDirection
    ) {}
}

File: Updog.Domain/Vote/Entities/Vote.cs
namespace Updog.Domain {
    /// <summary>
    /// A vote entity on a post, or comment.
    /// </summary>
    public sealed class Vote : Entity<Vote>, IUserEntity {
        #region Properties
        public int UserId { get; set; }
        public VotableEntityType ResourceType { get; set; }
        public int ResourceId { get; set; }
        public VoteDirection Direction { get; set; }
        #endregion

        #region Constructor(s)
        internal Vote(int id, int userId, VotableEntityType resourceType, int resourceId, VoteDirection direction) {
            Id = id;
            UserId = userId;
            ResourceType = resourceType;
            ResourceId = resourceId;
            Direction = direction;
        }

        internal Vote(int userId, VotableEntityType resourceType, int resourceId, VoteDirection direction) {
            UserId = userId;
            ResourceType = resourceType;
            ResourceId = resourceId;
            Direction = direction;
        }
        #endregion
    }
}